zero_bin/prover/src/lib.rs

```diff
@@ -83,20 +85,31 @@ impl BlockProverInput {
             save_inputs_on_error,
         };
 
-        // Segment the batches, prove segments and aggregate them to resulting batch
-        // proofs.
-        let batch_proof_futs: FuturesUnordered<_> = block_generation_inputs
+        let vec_segment_agg_proofs: FuturesUnordered<_> = block_generation_inputs
```

Comment:
@Nashtare @muursh 
I have separated segment proof and segment agg proof jobs here
Earlier, they were done in one step with map and fold
Though i made the changes but not sure if this there is a better way to do. Let me know your thoughts on this

Reply:
Hmm the fact that you added an await for `segment` only proofs is a bit concerning, I'm afraid we'll have a serious performance overhead by doing so, because we have many segments to aggregate together, and we can't do so ahead of finishing proving the segments.

@atanmarko do you have any idea on how to deal with this in a better way with the 2 different runtimes?

Reply:
https://github.com/0xPolygonZero/zk_evm/issues/627

Created this for handling this in future